### PR TITLE
[5.6] Support initial auto_increment

### PR DESCRIPTION
**Pull request by carusogabriel**
Inspired by @laraveldaily['s post](http://laraveldaily.com/set-auto-increment-start-laravel-migrations/) about set `auto-increment`, I've added that functionality to our `increments*` columns:
```php
$table->increments('id', 100);
```
Still working in other databases beyond `MySQL` :construction:

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please use camelCase for variable names.

**Comment by carusogabriel** [comment]
@GrahamCampbell but then we conflict with the existing one. Any suggestions? 😔

File: src/Illuminate/Database/Schema/Blueprint.php [hunk 1]
```diff
@@ -542,11 +547,12 @@ public function longText($column)
      * @param  string  $column
      * @param  bool  $autoIncrement
      * @param  bool  $unsigned
+     * @param  int  $auto_increment
      * @return \Illuminate\Support\Fluent
      */
-    public function integer($column, $autoIncrement = false, $unsigned = false)
+    public function integer($column, $autoIncrement = false, $unsigned = false, $auto_increment = 1)
```
Comment (carusogabriel) [inline src/Illuminate/Database/Schema/Blueprint.php]: @GrahamCampbell but then we conflict with the existing one. Any suggestions? 😔

**Comment by taylorotwell** [comment]
No plans to add right now.

**Comment by timothymarois** [comment]
@taylorotwell Was this never implemented? It should be. I never like to start items at 1 with incremental ids. 

You'll have to run a raw query like this: And that's a bit lame since we have a whole `Schema` dedicated to building these tables out. 

`DB::statement("ALTER TABLE table AUTO_INCREMENT = 1000;");`

I also don't think it should be on the `$table->increments()` method, but rather on the variable for table options like the use of `$table->engine = 'InnoDB';` which is on the same level. To follow by standard SQL options, it should be it's own property.

`$table->auto_increment = 1000;`

Those table options are described here, 
https://laravel.com/docs/5.7/migrations#tables

**Comment by rickmacgillis** [comment]
@taylorotwell - Any chance we can reopen this ticket, as it seems there's substantial interest in this feature?

**Comment by jangrimm** [comment]
> 
> 
> No plans to add right now.

Any good reasons underscoring the declination?

**Comment by clemjlt** [comment]
I agree with @timothymarois, it should be quite useful, especially used this way. Any plan/explication on its integration ?

**Comment by billwagner3** [comment]
I could really use this feature myself. Otherwise, it's going to be a real pain dealing with the database I need to import that has integrity issues we are trying to shore up. :(

File: src/Illuminate/Database/Schema/Blueprint.php (hunk 1 above)
Comment (pedzed) [reply src/Illuminate/Database/Schema/Blueprint.php]: I don't think it should be here anyway, but ` $autoIncrementValue`?

**Comment by pedzed** [comment]
@billwagner3 It's not going to be implemented (and rolled out) anytime soon. What is your workaround? I'm interested in other people's approaches.

**Comment by clemjlt** [comment]
@pedzed 
I've ended by doing it with a raw sql query in my migration : 
```php
class CreateUsersTable extends Migration
{
    public function up()
    {
        Schema::create('users', function (Blueprint $table) {
            $table->increments('id');
            $table->string('name');
            $table->timestamps();
        });
        DB::statement("ALTER TABLE users AUTO_INCREMENT = 10000;");
    }
}
```

**Comment by billwagner3** [comment]
> @billwagner3 It's not going to be implemented (and rolled out) anytime soon. What is your workaround? I'm interested in other people's approaches.

Because the users table PK only goes up to 300, I ended up inputting dummy data to fill all the holes, otherwise @pedzed I'd have used your approach. In the long run, this probably was the best solution for our situation.

**Comment by victor-ponamariov** [comment]
Would be useful feature :(

**Comment by armycoder** [comment]
It's going to be in 8.x

**Comment by DrAfridi** [comment]
This is coming in 8.x 😍

**Comment by prof-usman** [comment]
This is coming in 8.x tomorrow 😍

**Comment by achyutkneupane** [comment]
**Is this feature now in Laravel 8.x?**

**Comment by clemjlt** [comment]
Yes, you can do it this way :

```public function up()
{
    Schema::create('posts', function (Blueprint $table) {
        $table->id()->startingValue(1200);
        ...
```

**Comment by achyutkneupane** [comment]
> Yes, you can do it this way :
> 
> ```
> {
>     Schema::create('posts', function (Blueprint $table) {
>         $table->id()->startingValue(1200);
>         ...
> ```

Thankyou [https://github.com/clemjlt](https://github.com/clemjltA)

**Comment by achyutkneupane** [comment]
> Yes, you can do it this way :
> 
> ```
> {
>     Schema::create('posts', function (Blueprint $table) {
>         $table->id()->startingValue(1200);
>         ...
> ```

Thankyou [clemjlt](https://github.com/clemjlt)

**Comment by codetestmg** [comment]
Laravel : 9.19
PHP : 8.1.6


`$table->id()->startingValue(50)->change();`
modifying the existing table doesn't work.Only if fresh migration runs then it works.
any work around for this

**Pull request closed without merging**